modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellTest.java

```diff
@@ -833,5 +836,124 @@ public void testListCellHeights() {
+        list.setOnEditStart(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(initialCellIndex);
+        list.edit(listEditingIndex);
+        assertEquals("sanity: list editingIndex ", listEditingIndex, list.getEditingIndex());
+        assertFalse("sanity: cell must not be editing", cell.isEditing());
+        cell.updateIndex(listEditingIndex);
+        assertEquals("sanity: index updated ", listEditingIndex, cell.getIndex());
+        assertEquals("list editingIndex unchanged by cell", listEditingIndex, list.getEditingIndex());
+        assertTrue(cell.isEditing());
+        assertEquals(1, events.size());
+    }
+
+    @Test
+    public void testChangeIndexOffEditing0_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(1, 0);
+    }
+    @Test
+    public void testChangeIndexOffEditing1_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(1, -1);
+    }
+    @Test
+    public void testChangeIndexOffEditing2_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(0, 1);
+    }
+    @Test
+    public void testChangeIndexOffEditing3_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(0, -1);
+    }
+
+    public void assertUpdateCellIndexOffEditing(int editingIndex, int cancelIndex) {
+        list.getFocusModel().focus(-1);
+        List<EditEvent> events = new ArrayList<EditEvent>();
+        list.setOnEditCancel(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(editingIndex);
+        list.edit(editingIndex);
+        assertEquals("sanity: list editingIndex ", editingIndex, list.getEditingIndex());
+        assertTrue("sanity: cell must be editing", cell.isEditing());
+        cell.updateIndex(cancelIndex); // change cell index to negative
+        assertEquals("sanity: index updated ", cancelIndex, cell.getIndex());
+        assertEquals("list editingIndex unchanged by cell", editingIndex, list.getEditingIndex());
+        assertFalse("cell must not be editing if cell index is " + cell.getIndex(), cell.isEditing());
+        assertEquals(1, events.size());
+    }
+
+    @Test
+    public void testMisbehavingCancelEditTerminatesEdit() {
+        ListCell<String> cell = new MisbehavingOnCancelListCell<>();
+
+        list.setEditable(true);
+        cell.updateListView(list);
+
+        int editingIndex = 1;
+        int intermediate = 0;
+        int notEditingIndex = -1;
+        cell.updateIndex(editingIndex);
+        list.edit(editingIndex);
+        assertTrue("sanity: ", cell.isEditing());
+        try {
+            list.edit(intermediate);
+        } catch (Exception ex) {
+            // just catching to test in finally
+        } finally {
+            assertFalse("cell must not be editing", cell.isEditing());
+            assertEquals("table must be editing at intermediate index", intermediate, list.getEditingIndex());
+        }
```

Comment:
that's good - except for a minor omission: the catch-block is never reached, instead it is printed to sysout (or syserr, don't nail me) That happens because ExpressionHelper kind of swallows all exceptions in change notifications by passing them to the uncaughtExceptionHandler. To fix this, a test should un-/install (in After/Before methods) an uncaught exception handler. For examples, see f.i. TableCellTest.

really minor: replace "table" by "list" in the failure message :)

Reply:
I've now added the exception handler, like in TreeCellTest.

And I've fixed the texts of the failure messages.